### PR TITLE
Add flake8 config file

### PR DESCRIPTION
**Pull request by danielmellado**
##### SUMMARY
Add flake8 config file

Even if ansible-test is handling pep8 testing, this commit brings back
and updates the missing flake8 config that went away when removing
legaxy tox.ini.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
flake8

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
```
Comment (gundalow) [inline .flake8:1]: I wonder if we need a comment in here (and in ansible-test's config) to remind people to keep these two files in sync

**Comment by gundalow** [comment]
Please get wait for approval from Matt Clay, incase there is something else we are not aware of.

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 31e6fdf5c49ebf621574a700b5b932da5184de58

File: test/sanity/code-smell/package-data.py [hunk 2]
```diff
@@ -61,6 +61,7 @@ def assemble_files_to_ship(complete_file_list):
         'hacking/test-module',
         'hacking/test-module.py',
         '.cherry_picker.toml',
+        '.flake8',
```
Comment (webknjaz) [inline test/sanity/code-smell/package-data.py:64]: You may need to include this into https://github.com/ansible/ansible/blob/devel/MANIFEST.in

Comment (danielmellado) [reply test/sanity/code-smell/package-data.py:64]: I don't really think so, for the same reason about why we don't include `.cherry_picker.toml`. Summarizing, it's a dev only thing which wouldn't really need to make its way to any installation nor package.

File: .flake8 (hunk 1 above)
Comment (danielmellado) [reply .flake8:1]: yeah, I think it'd be helpful not now but in the future. Honestly I'd love for pep8 to be able to map its ignores to a list, pretty much like a `requirements.txt`

**Comment by abadger** [comment]
I'm not sure we want this for the same reason tox.ini was removed.  I don't include config for things we don't use.  They get out of date wrt the config for the things we do use and people assume that the lint tool should run without errors when we can't say that because we don't actively maintain it.  @mattclay will need to make that decision when he gets back from pto, though.

**Comment by danielmellado** [comment]
seems shippable had some issues with docker, closing and reopening to retrigger

**Comment by mattclay** [comment]
Configuration should only be included for tools we're using. If there is interest in enforcing flake8 rules in CI then it should be added as a topic to the core meeting agenda so it can be discussed.

We would need to decide whether or not it is useful, especially given existing tests, as well as what configuration to use if it were enabled.

**Comment by mattclay** [comment]
Closing this since we don't want to add configuration for tools we're not using.

**Pull request closed without merging**